### DISCUSSION TITLE
[@types/node] typing for EventEmitterAsyncResource et al. missing

### DISCUSSION THREAD

**Discussion by KilianKilmister** [DefinitelyTyped/DefinitelyTyped#61365] [Issues with a @types package] [answered]
The `EventEmitterAsyncResource` class (Added in: v17.4.0, v16.14.0) and related types are missing in the `@types/node`.

[link to nodejs doc](https://nodejs.org/api/events.html#class-eventseventemitterasyncresource-extends-eventemitter)

I typed them out and made a [gist], but I don't understand the typing of the `events` module in `@types/node`, so I wouldn't know how to fix them. The whole `import /* name */ = require(/* module */)` and `export = /* name */` thing.

The gist also includes types for the missing `EventEmitter.prototype[Symbol.for('nodejs.rejection')]` method.

[link to nodejs doc](https://nodejs.org/api/events.html#emittersymbolfornodejsrejectionerr-eventname-args)

I've typed out a bunch of these smaller things over time, but I don't have the spare time to contribute a PR to DefinitlyTyped, and I really don't understand the structure of parts of the `@types/node` package, but I'll gladly provide the typings+doc-comments here in the discussions if someone's willing and able to integrate them into the package.

[gist]:<https://gist.github.com/KilianKilmister/d5b3474610787dac35847c4e795625da>

[a comment by an automated account is not shown]

**Comment by jerome-benoit** [comment]
@KilianKilmister: Please do PR to make it happens.

**Comment by peterblazejewicz** [comment, marked as answer]
fixed in #66814
